gio/src/auto/initable.rs

```diff
@@ -0,0 +1,67 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use crate::Cancellable;
+use glib::object::IsA;
+use glib::translate::*;
+use std::fmt;
+use std::ptr;
+
+glib::wrapper! {
+    #[doc(alias = "GInitable")]
+    pub struct Initable(Interface<ffi::GInitable, ffi::GInitableIface>);
+
+    match fn {
+        type_ => || ffi::g_initable_get_type(),
+    }
+}
+
+impl Initable {
+    //#[doc(alias = "g_initable_new")]
+    //pub fn new<P: IsA<Cancellable>>(object_type: glib::types::Type, cancellable: Option<&P>, error: &mut glib::Error, first_property_name: Option<&str>, : /*Unknown conversion*//*Unimplemented*/Fundamental: VarArgs) -> glib::Object {
+    //    unsafe { TODO: call ffi:g_initable_new() }
+    //}
+
+    //#[doc(alias = "g_initable_new_valist")]
+    //pub fn new_valist<P: IsA<Cancellable>>(object_type: glib::types::Type, first_property_name: &str, var_args: /*Unknown conversion*//*Unimplemented*/Unsupported, cancellable: Option<&P>) -> Result<glib::Object, glib::Error> {
+    //    unsafe { TODO: call ffi:g_initable_new_valist() }
+    //}
+
+    //#[cfg_attr(feature = "v2_54", deprecated = "Since 2.54")]
+    //#[doc(alias = "g_initable_newv")]
+    //pub fn newv<P: IsA<Cancellable>>(object_type: glib::types::Type, parameters: /*Ignored*/&[&glib::Parameter], cancellable: Option<&P>) -> Result<glib::Object, glib::Error> {
+    //    unsafe { TODO: call ffi:g_initable_newv() }
+    //}
+}
+
+pub const NONE_INITABLE: Option<&Initable> = None;
+
+pub trait InitableExt: 'static {
+    #[doc(alias = "g_initable_init")]
+    fn init<P: IsA<Cancellable>>(&self, cancellable: Option<&P>) -> Result<(), glib::Error>;
+}
+
+impl<O: IsA<Initable>> InitableExt for O {
+    fn init<P: IsA<Cancellable>>(&self, cancellable: Option<&P>) -> Result<(), glib::Error> {
```

Comment:
Calling this multiple times is not necessarily safe AFAIU from the docs

Reply:
So maybe this needs to be `unsafe` (and that's why having a safe `new()` would be useful). Not entirely sure.

Reply:
Yes, it's not safe to be called more than once (or anything other than exactly once).
It's not clear to me however how to enforce that: in pure Rust I would have used a type-state / builder-object pattern kind of thing where the init takes ownership of self and returns a type which does not support Initable anymore, but in this interop scenario with the gobject model I'm not sure of the approach to take.

Reply:
You mark it `unsafe` and add a `new()` that internally calls it :)

Reply:
I wrote my message in between your first two and didn't refresh :) it was actually pretty clear what to do

Reply:
Done. I called it `initable_new` to avoid name conflicts with `new` which is in widespread use, but I can easily change that.